File: ChatGPT/Direct_Attempt/python/output/output_19b.py
# ChatGPT-generated solution will go here.
def main():
    with open("input.txt") as f:
        sections = f.read().strip().split("\n\n")

    towel_patterns = sections[0].split(", ")
    designs = sections[1].splitlines()
    towel_set = set(towel_patterns)

    from functools import lru_cache

    @lru_cache(maxsize=None)
    def count_ways(design):
        if design == "":
            return 1
        total = 0
        for pattern in towel_set:
            if design.startswith(pattern):
                total += count_ways(design[len(pattern):])
        return total

    total_ways = sum(count_ways(design) for design in designs)
    print(total_ways)

if __name__ == "__main__":
    main()

File: ChatGPT/Direct_Attempt/clojure/output/output_19b.clj
;; ChatGPT-generated solution will go here.
(ns output_19b
  (:require [clojure.string :as str]))

(defn parse-input [lines]
  (let [[patterns-line _ & designs] lines
        patterns (str/split patterns-line #",\s*")]
    {:patterns patterns
     :designs designs}))

(defn count-ways [patterns design]
  (let [n (count design)
        pattern-set (set patterns)
        dp (long-array (inc n) 0)]
    (aset dp 0 1)
    (doseq [i (range 1 (inc n))]
      (doseq [j (range (max 0 (- i 10)) i)]
        (when (pattern-set (subs design j i))
          (aset dp i (+ (aget dp i) (aget dp j))))))
    (aget dp n)))

(defn part [lines]
  (let [{:keys [patterns designs]} (parse-input lines)]
    (reduce + (map #(count-ways patterns %) designs))))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
